frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -3653,6 +3658,7 @@ const IR::Node *TypeInference::postorder(IR::MethodCallExpression *expression) {
                 LOG3("Folding " << mem << " to " << w);
                 if (w < 0) return expression;
                 if (mem->member.name.endsWith("Bytes")) w = ROUNDUP(w, 8);
+                if (getParent<IR::MethodCallStatement>()) return nullptr;
```

Comment:
This exits early before the type is set, could this have any consequences?

Reply:
Why do we need to set a type for a node that is being removed?

Reply:
I guess this change only applies for the size and valid functions so it should be safe.